bundles/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/internal/handler/PentairIntelliChlorHandler.java

```diff
@@ -34,91 +37,155 @@
  *
  * @author Jeff James - Initial contribution
  */
+@NonNullByDefault
 public class PentairIntelliChlorHandler extends PentairBaseThingHandler {
     private final Logger logger = LoggerFactory.getLogger(PentairIntelliChlorHandler.class);
 
-    protected PentairPacketIntellichlor pic3cur = new PentairPacketIntellichlor();
-    protected PentairPacketIntellichlor pic4cur = new PentairPacketIntellichlor();
+    protected int version;
+    protected String name = "";
+
+    /** for a saltoutput packet, represents the salt output percent */
+    public int saltOutput;
+    /** for a salinity packet, is value of salinity. Must be multiplied by 50 to get the actual salinity value. */
+    public int salinity;
+
+    public boolean ok;
+    public boolean lowFlow;
+    public boolean lowSalt;
+    public boolean veryLowSalt;
+    public boolean highCurrent;
+    public boolean cleanCell;
+    public boolean lowVoltage;
+    public boolean lowWaterTemp;
+    public boolean commError;
+
+    @Nullable
+    public static PentairIntelliChlorHandler onlineChlorinator;
```

Comment:
Why do you need a this static field?

Reply:
Yes, this is used to determine if there is already a chlorinator initialized and online.  The binding only supports a single chlorinator.

Reply:
Why is only a single chlorinator allowed? Is it not possible for people to have multiple on their system?

Reply:
Correct, the pool system can only control one chlorinator in the system.  The protocol for the intellichlor is unique from the standard pentair protocol and there is no mechanism to control more than one.